src/pudl/output/eia923.py

```diff
@@ -248,29 +253,32 @@ def fuel_receipts_costs_eia923(pudl_engine, freq=None,
         logger.info('filling in fuel cost NaNs EIA APIs monthly state averages')
         fuel_costs_avg_eiaapi = get_fuel_cost_avg_eiaapi(
             FUEL_COST_CATEGORIES_EIAAPI)
-        # add the state from the plants table
-        frc_df = (
-            pudl.helpers.merge_on_date_year(
-                frc_df,
-                pudl.output.eia860.plants_eia860(
-                    pudl_engine, start_date=start_date, end_date=end_date)[
-                        ['report_date', 'plant_id_eia', 'state']],
-                on=['plant_id_eia', ], how='left')
-            .merge(fuel_costs_avg_eiaapi,
-                   on=['report_date', 'state', 'fuel_type_code_pudl'],
-                   how='left')
-            .assign(
-                # add a flag column to note if we are using the api data
-                fuel_cost_from_eiaapi=lambda x:
-                np.where(x.fuel_cost_per_mmbtu.isnull()
-                         & x.fuel_cost_per_unit.notnull(),
-                         True, False),
-                fuel_cost_per_mmbtu=lambda x:
-                np.where(x.fuel_cost_per_mmbtu.isnull(),
-                         (x.fuel_cost_per_unit
-                          / x.heat_content_mmbtu_per_unit),
-                         x.fuel_cost_per_mmbtu)
-            )
+        # Merge to bring in states associated with each plant:
+        plant_states = pd.read_sql(
+            "SELECT plant_id_eia, state FROM plants_entity_eia;", pudl_engine
+        )
+        frc_df = frc_df.merge(plant_states, on="plant_id_eia", how="left")
+
+        # Merge in monthly per-state fuel costs from EIA based on fuel type.
+        frc_df = frc_df.merge(
+            fuel_costs_avg_eiaapi,
+            on=['report_date', 'state', 'fuel_type_code_pudl'],
```

Comment:
Yes, at this point we know that `frc_df` is always monthly (because it's fresh from the DB) and we know that the EIA data is monthly (because that's how it's published in the EIA API).